upgrades/raft.go

```diff
@@ -23,6 +24,13 @@ const jujuMachineKey = "juju-machine-id"
 // being upgraded.
 func BootstrapRaft(context Context) error {
 	agentConfig := context.AgentConfig()
+	storageDir := filepath.Join(agentConfig.DataDir(), "raft")
```

Comment:
Do you think that just looking for the directory is sufficient?

Reply:
Yes - that directory is created in the process of creating the logstore or snapshot store, immediately before the bootstrapping. The only way the directory could exist without the raft cluster having been bootstrapped is if there was catastrophic error while bootstrapping, in which case the bootstrap will have failed (and the controller would have been torn down or the upgrade will have failed).